cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/golang/GoLanguage.kt

```diff
@@ -46,21 +43,62 @@ open class GoLanguage : Language<GoLanguageFrontend>(), HasShortCircuitOperators
     override val startCharacter = '['
     override val endCharacter = ']'
 
+    /** See [Documentation](https://pkg.go.dev/builtin). */
     @Transient
     override val simpleTypes =
         mapOf(
+            // https://pkg.go.dev/builtin#any
+            // TODO: Actually, this should be a type alias to interface{}
+            "any" to ObjectType("any", listOf(), false, this),
```

Comment:
Are these ObjectTypes really "simple types"? Is it like `java.lang.Object` or so?

Reply:
That's complicated :) Actually it is a so called "interface type", a concept which is probably unique to Go. Where as an ObjectType is describing something that is of a particular class (or Java interface, which is also just a name basically), the "interface type" in Go says, this object is something that implements this specific interface, e.g. a set of functions. And it's really dependent on the functions you specify in this interface and not the name, the name is totally irrelevant actually. This is very close to the duck typing concept in ruby, but probably unique to those two languages so not quite 100 % sure how we should model it.

In the end we could have an interface type that has a list of function types describing it. So the ingredients would be there, I am just not sure if we want to have it. And also whether we then want to for example model Java "interfaces" as an "InterfaceType" or an "ObjectType".

Reply:
Ok. Let's keep it like this